libs/actions/sendmail.c

```diff
@@ -0,0 +1,226 @@
+		return -1;
+	} 
+	if(settings_find_string("smtp-password", &stmp) != EXIT_SUCCESS) {
+		logprintf(LOG_ERR, "Sendmail: setting \"smtp-password\" is missing in config");
+		return -1;
+	} 
+	if(settings_find_string("mail-to", &stmp) != EXIT_SUCCESS) {
+		logprintf(LOG_ERR, "Sendmail: setting \"mail-to\" is missing in config");
+		return -1;
+	} 
+	if(settings_find_string("mail-from", &stmp) != EXIT_SUCCESS) {
+		logprintf(LOG_ERR, "Sendmail: setting \"mail-from\" is missing in config");
+		return -1;
+	} 
+	if(settings_find_string("mail-subject", &stmp) != EXIT_SUCCESS) {
+		logprintf(LOG_ERR, "Sendmail: setting \"mail-subject\" is missing in config");
+		return -1;
+	} 
+	return 0;
+}
+
+static int actionSendmailRun(struct JsonNode *arguments) {
+	struct JsonNode *jsubject = NULL;
+	struct JsonNode *jmessage = NULL;
+	struct JsonNode *jto = NULL;
+	struct JsonNode *jvalues1 = NULL;
+	struct JsonNode *jvalues2 = NULL;
+	struct JsonNode *jvalues3 = NULL;
+	struct JsonNode *jval1 = NULL;
+	struct JsonNode *jval2 = NULL;
+	struct JsonNode *jval3 = NULL;
+	struct mail_t mail;
+	char *message = NULL, *to = NULL, *subject = NULL;
+	char *sfrom = NULL, *sto = NULL, *ssubject = NULL;
+	char *shost = NULL, *suser = NULL, *spassword = NULL;
+	int sport= 0;	
+
+	// mail settings
+	settings_find_string("mail-from", &sfrom);
+	settings_find_string("mail-to", &sto);
+	settings_find_string("mail-subject", &ssubject);
+
+	//smtp settings
+	settings_find_string("smtp-host", &shost);
+	settings_find_number("smtp-port", &sport);
+	settings_find_string("smtp-user", &suser);
+	settings_find_string("smtp-password", &spassword);
+
+	// arguments
+	jsubject = json_find_member(arguments, "SUBJECT");
+	jmessage = json_find_member(arguments, "MESSAGE");
+	jto = json_find_member(arguments, "TO");
+	
+	if(jmessage != NULL) {	
+		if((jvalues1 = json_find_member(jmessage, "value")) != NULL) {
+			if((jval1 = json_find_element(jvalues1, 0)) != NULL && jval1->tag == JSON_STRING){
+				message = jval1->string_;
+			}
+			subject = ssubject;
+			if(jsubject != NULL) {	
+				if((jvalues2 = json_find_member(jsubject, "value")) != NULL) {
+					if((jval2 = json_find_element(jvalues2, 0)) != NULL && jval2->tag == JSON_STRING){
+						subject = jval2->string_;
+					} 
+				}
+			}
+			to = sto;
+			if(jto != NULL) {	
+				if((jvalues3 = json_find_member(jto, "value")) != NULL) {
+					if((jval3 = json_find_element(jvalues3, 0)) != NULL && jval3->tag == JSON_STRING){
+						to = jval3->string_;
+					}
+				}
+			}
+			
+			mail.from = MALLOC(strlen(sfrom)+1);
+			mail.to = MALLOC(strlen(to)+1);
+			mail.subject = MALLOC(strlen(subject)+1);
+			mail.message = MALLOC(strlen(message)+1);
+
+			strcpy(mail.from, sfrom);
+			strcpy(mail.to, to);
+			strcpy(mail.subject, subject);
+			strcpy(mail.message, message);
+			
+			if(sendmail(shost, suser, spassword, sport, &mail) != 0) {
+				logprintf(LOG_DEBUG, "Sendmail failed with message \"%s\"", message);
```

Comment:
I assume the message here is some sort of error message, but it isn't. So why the error like this?